src/components/crossDevice/CrossDeviceLink/index.js

```diff
@@ -77,50 +82,134 @@ class CrossDeviceLink extends Component {
   }
 
   render = () =>
-    this.props.roomId ? <CrossDeviceLinkUI roomId={this.props.roomId} /> : <Spinner />
+    this.props.roomId ? <CrossDeviceLinkUI {...this.props}/> : <Spinner />
 }
 
 class CrossDeviceLinkUI extends Component {
   constructor(props) {
     super(props)
-    this.state = {copySuccess: false}
+    this.state = {
+      copySuccess: false,
+      sending: false,
+      mobileNumber: null,
+      error: {}
+    }
   }
 
+  linkCopiedTimeoutId = null
+
   copyToClipboard = (e) => {
+    e.preventDefault()
     this.textArea.select()
     document.execCommand('copy')
     e.target.focus()
     this.setState({copySuccess: true})
+    this.clearLinkCopiedTimeout()
+    this.linkCopiedTimeoutId = setTimeout(() => {
+      this.setState({copySuccess: false})
+    }, 5000)
+  }
+
+  clearLinkCopiedTimeout = () => {
+    if (this.linkCopiedTimeoutId) {
+      clearTimeout(this.linkCopiedTimeoutId)
+    }
+  }
+
+  updateNumber = (mobileNumber) => {
+    this.setState({mobileNumber})
+  }
+
+  clearPreviousAttempts = () => {
+    if (this.hasNumberError()) this.clearError()
+    if (this.state.mobileNumber) this.clearNumber()
+  }
+
+  clearNumber = () => {
+    this.setState({mobileNumber: null})
+  }
+
+  clearError = () => {
+    this.setState({error: {}})
+  }
+
+  setError = (name, type) => this.setState({error: {name, type}})
+
+  handleResponse = (response) => {
+    this.setState({sending: false})
+    if (response.status === "OK") {
+      this.props.actions.setMobileNumber(this.state.mobileNumber)
+      this.props.nextStep()
+    }
+    else {
+      this.setError('SMS_FAILED', 'error')
+    }
+  }
+
+  handleSMSError = ({status}) => {
+    this.setState({sending: false})
+    status === 429 ? this.setError('SMS_OVERUSE', 'error') : this.setError('SMS_FAILED', 'error')
+  }
+
+  hasNumberError = () => this.state.error.name === 'numberError'
+
+  sendSms = () => {
+    this.clearError()
+    if (this.state.mobileNumber) {
+      this.setState({sending: true})
+      const options = {
+        payload: JSON.stringify({to: this.state.mobileNumber, id: this.props.roomId}),
```

Comment:
You aren't sending the full path. How will the server know which base 36 version to use?

Reply:
In the telephony documentation there was written that you need to include the roomId, not the path. I will add the path if that's what needed but docs should be updated too.

Reply:
Oh, I can see docs are correct